app/Repositories/BaseRepository.php

```diff
@@ -0,0 +1,343 @@
+        );
+    }
+
+    /**
+     * @param  int $id
+     * @return mixed
+     */
+    public function getById($id)
+    {
+        return $this->model->findOrFail($id);
+    }
+
+    /**
+     * @param  array $columns
+     * @return mixed
+     */
+    public function all($orderBy = 'updated_at desc', $columns = ['*'])
+    {
+        return $this->model->orderByRaw($orderBy)->get($columns);
+    }
+
+    /**
+     * @param  int   $perPage Per page
+     * @param  array $columns Array Columns
+     * @return array Return data paginate
+     */
+    public function paginate($perPage = 15, $columns = ['*'])
+    {
+        return $this->model->paginate($perPage, $columns);
+    }
+
+    /**
+     * @param  array $data Array data
+     * @return void
+     */
+    public function create(array $data)
+    {
+        return $this->model->create($data);
+    }
+
+    /**
+     * @param  array  $data      Array data
+     * @param  $id ID
+     * @param  string $attribute Attribute
+     * @return void
+     */
+    public function update(array $data, $id, $attribute = "id")
+    {
+         return $this->model->where($attribute, '=', $id)->update($data);
+    }
+
+    /**
+     * @param  array      $data      Array data
+     * @param  $id ROLE_ID
+     * @param  string     $attribute Attribute
+     * @return void
+     */
+    public function update_role(array $data, $role_id, $attribute = "role_id")
+    {
+         return $this->model->where($attribute, '=', $role_id)->update($data);
+    }
+
+    /**
+     * @param  $id ID
+     * @return void
+     */
+    public function delete($id)
+    {
+        return $this->model->destroy($id);
+    }
+
+    /**
+     * @param  $id ID
+     * @param  array $columns Columns
+     * @return array Return array
+     */
+    public function find($id, $columns = ['*'])
+    {
+        return $this->model->find($id, $columns);
+    }
+
+    /**
+     * @param  $attribute Attribute
+     * @param  $value Value
+     * @param  array               $columns Columns
+     * @return array Array
+     */
+    public function findBy($attribute, $value, $columns = ['*'])
+    {
+        return $this->model->where($attribute, '=', $value)->first($columns);
+    }
+
+    public function findByReturnArray($attribute, $value, $columns = ['*'])
+    {
+        return $this->model->where($attribute, '=', $value)->get($columns);
+    }
+
+    public function findByReturnArrayPlug($attribute, $value, $pluck, $columns = ['*'])
+    {
+        return $this->model->where($attribute, '=', $value)->get($columns)->pluck($pluck);
+    }
+    public function findByMultiConditionsLimit($whereData = [], $columns = ['*'])
+    {
+        return $this->model->where($whereData)->first($columns);
+    }
+    /**
+     * @param  $tableName Table name
+     * @param  array                $whereData Array where data
+     * @return array Return array
+     */
+    public function findByMultiConditions($tableName, $whereData = [])
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Strings.DoubleQuoteUsage.NotRequired</br>**Severity**: error</br>**File**: app/Repositories/BaseRepository.php</br>**Line**: 243</br>String "*" does not require double quotes; use single quotes instead